resolver_test.go

```diff
@@ -35,3 +35,64 @@ func TestExhaustiveErrors(t *testing.T) {
 
 	assert.JSONEq(t, `{"title":"Bad Request","status":400,"detail":"Error while parsing input parameters","errors":[{"message":"cannot parse boolean","location":"query.bool","value":"bad"},{"message":"cannot parse integer","location":"query.int","value":"bad"},{"message":"cannot parse float","location":"query.float32","value":"bad"},{"message":"cannot parse float","location":"query.float64","value":"bad"},{"message":"cannot parse integer","location":"query[2].tags","value":"bad"},{"message":"unable to validate against schema: invalid character 'b' looking for beginning of value","location":"query.tags","value":"[1,2,bad]"},{"message":"cannot parse time","location":"query.time","value":"bad"},{"message":"Must be greater than or equal to 5","location":"body.value","value":1}]}`, w.Body.String())
 }
+
+type Dep1 struct {
+	// Only *one* of the following two may be set.
+	One string `json:"one,omitempty"`
+	Two string `json:"two,omitempty"`
+}
+
+func (d *Dep1) Resolve(ctx Context, r *http.Request) {
+	if d.One != "" && d.Two != "" {
+		ctx.AddError(&ErrorDetail{
+			Message:  "Only one of ['one', 'two'] is allowed.",
+			Location: "one",
+			Value:    d.One,
+		})
+	}
+}
+
+type Dep2 struct {
+	// Test recursive resolver with complex input structure containing a map of
+	// lists of struct pointers.
+	Foo map[string][]*Dep1 `json:"foo"`
+}
+
+func TestNestedResolver(t *testing.T) {
+	app := newTestRouter()
+
+	app.Resource("/").Post("test", "Test",
+		NewResponse(http.StatusNoContent, "desc"),
+	).Run(func(ctx Context, input struct {
+		Body Dep2
+	}) {
+		ctx.WriteHeader(http.StatusNoContent)
+	})
+
+	// Test happy case just sending ONE of the two possible fields in each struct.
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest(http.MethodPost, "/", strings.NewReader(`{"foo": {"a": [{"one": "1"}], "b": [{"two": "2"}]}}`))
+	app.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusNoContent, w.Result().StatusCode)
```

Comment:
Do we want to validate what the resolve returns?

Reply:
With the simplified test it's just pass/fail so I think checking for the expected return code is enough here. We aren't computing any values anymore.